### PR TITLE
add tt-xla tutorial, deprecate torch tutorials

### PR DESCRIPTION
**Pull request by einnesTT**
Adding a new tutorial that provides a basic template for compiling a model with TT-XLA using PyTorch or JAX. 

Deprecated torch tutorial folder. Torch tutorials will move to TT-XLA after a batch normalization op is added in TT-MLIR. (Provided timeline from Defne is that the op will be added some time next week.)

### CODE REVIEW

**Comment by ddilbazTT** [comment]
Just for clarity, we are holding off generative model demos in tt-xla since they are blocked by `batch_norm_training` op.  https://github.com/tenstorrent/tt-mlir/issues/5162

**Comment by einnesTT** [comment]
Yes - those are on there waiting until next week, then I'll move them out with the fixes that make them usable for tt-xla. They worked for tt-torch, but the tt-torch folder is now listed as deprecated.

File: tt-forge/tt-xla/compile_models_single_chip.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+When writing code for compiling a model for use with TT-XLA, if you are using PyTorch, you can use the [Torch-XLA APIs](https://docs.pytorch.org/xla/release/r2.8/learn/pytorch-on-xla-devices.html) and Tenstorrent extensions. Tenstorrent uses Torch-XLA to emit XLA HLO and then TT-XLA compiles it. 
+
+The general structure is: 
+
+```python
+import torch_xla.core.xla_model as xm
+import torch_xla.runtime as xr
+
+# By default torch_xla uses the CPU device so set it to the TT device.
+xr.set_device_type("TT")
```
Comment (jameszianxuTT) [inline tt-forge/tt-xla/compile_models_single_chip.md:39]: I think it would be good to note that this must be run before acquiring a device.

File: tt-forge/tt-xla/compile_models_single_chip.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+| import torch_xla.runtime as xr | Imports the Tenstorrent specific runtime module, required to work with Tenstorrent devices. |
+| xr.set_device_type("TT") | Sets the target device to Tenstorrent hardware ("TT"). |
+| options = {...} | Defines a dictionary of TT-XLA compiler flags.|
+| "enable_optimizer": "true" | A TT-XLA compiler flag, it goes in `options = {...}`. Enables TT-XLA compiler optimizations on the computation graph. |
+| "enable_memory_layout_analysis": "true" | A TT-XLA compiler flag, it goes in `options = {...}`. It
+allows for memory layout tuning. |
+| "enable_l1_interleaved": "true" | A TT-XLA compiler flag, it goes in `options = {...}`. It boosts memory access efficiency across cores. |
+| "enable_fusing_conv2d_with_multiply_pattern": "true" | A TT-XLA compiler flag, it goes in `options = {...}`. Enables fusion of Conv2D followed by Multiply into a single operation to reduce op count and memory movement. |
+| torch_xla.set_custom_compile_options(options) | This command globally registers any compiler options you select| 
+| model.compile(backend="tt") | A Tenstorrent specific method for compiling models. |
```
Comment (jameszianxuTT) [inline tt-forge/tt-xla/compile_models_single_chip.md:85]: I would change this to something like this:

Equivalent to calling torch.compile on an torch.nn.Module "model" with the "tt" backend.

File: tt-forge/tt-xla/compile_models_single_chip.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+# Connect the device.
+device = xm.xla_device() 
+
+# Define compiler options. (You do not need to include all of these.)
+options = {
+    "enable_optimizer": "true",
+    "enable_memory_layout_analysis": "true",
+    "enable_l1_interleaved": "true",
+    "enable_fusing_conv2d_with_multiply_pattern": "true",
```
[the remark this answers is not in the document]
Comment (einnesTT) [reply tt-forge/tt-xla/compile_models_single_chip.md:49]: I added the comments from that file to the chart for the code sample.

File: tt-forge/tt-xla/compile_models_single_chip.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+# Move output to CPU. 
+output = output.to("cpu")
+```
+
+The following chart lists the aspects of the template that are Tenstorrent specific: 
+
+| Code | Explanation |
+|------|-------------|
+| import torch_xla.runtime as xr | Imports the Tenstorrent specific runtime module, required to work with Tenstorrent devices. |
```
[the remark this answers is not in the document]
Comment (einnesTT) [reply tt-forge/tt-xla/compile_models_single_chip.md:76]: Removed from the chart

File: tt-forge/tt-xla/compile_models_single_chip.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+| xr.set_device_type("TT") | Sets the target device to Tenstorrent hardware ("TT"). |
+| options = {...} | Defines a dictionary of TT-XLA compiler flags.|
+| "enable_optimizer": "true" | A TT-XLA compiler flag, it goes in `options = {...}`. Enables TT-XLA compiler optimizations on the computation graph. |
+| "enable_memory_layout_analysis": "true" | A TT-XLA compiler flag, it goes in `options = {...}`. It
+allows for memory layout tuning. |
+| "enable_l1_interleaved": "true" | A TT-XLA compiler flag, it goes in `options = {...}`. It boosts memory access efficiency across cores. |
+| "enable_fusing_conv2d_with_multiply_pattern": "true" | A TT-XLA compiler flag, it goes in `options = {...}`. Enables fusion of Conv2D followed by Multiply into a single operation to reduce op count and memory movement. |
+| torch_xla.set_custom_compile_options(options) | This command globally registers any compiler options you select|
```
[the remark this answers is not in the document]
Comment (einnesTT) [reply tt-forge/tt-xla/compile_models_single_chip.md:84]: Corrected.

File: tt-forge/tt-xla/compile_models_single_chip.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+# Optional: serialize the compiled artifacts for deployment or inspection.
+serialize_compiled_artifacts_to_disk(model, input_data, output_prefix="output/model")
+```
+
+The following chart lists the aspects of the template that are Tenstorrent specific: 
+
+| Code | Explanation |
+|---|---|
+| from tt_jax import serialize_compiled_artifacts_to_disk | Imports a function from Tenstorrent's tt_jax module that takes your JAX model and compiles it for Tenstorrent hardware. |
```
[the remark this answers is not in the document]
Comment (einnesTT) [reply tt-forge/tt-xla/compile_models_single_chip.md:119]: Ok

File: tt-forge/tt-xla/compile_models_single_chip.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
+# Compiling Models in PyTorch and JAX Using TT-XLA (Single Chip)
+
+This tutorial explains how to write code to compile models for execution on Tenstorrent hardware using the TT-XLA frontend. TT-XLA is Tenstorrent's primary open-source compiler for machine learning that takes computation graphs from PyTorch and JAX frameworks, applies optimizations, and compiles them into optimized, low-level machine code for Tenstorrent chips.
+
+The TT-XLA frontend supports: 
+
+* **PyTorch** - A robust, flexible framework for building, training, and deploying neural networks and other machine learning models. 
+* **JAX** - A Python framework that combines a NumPy-like API for array manipulation with composable function transformations for automatic differentiation, vectorization, and just-in-time (JIT) compilation using XLA for acceleration on GPUs and TPUs.
```
[the remark this answers is not in the document]
Comment (einnesTT) [reply tt-forge/tt-xla/compile_models_single_chip.md:8]: I could say 'especially on Tenstorrent hardware' instead?

File: tt-forge/tt-xla/compile_models_single_chip.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+# Define a simple JAX function.
+def model(x):
+    return jnp.sin(x) * jnp.cos(x)
+
+# Create example input data.
+input_data = jnp.ones((10,))
+
+# Compile the function for the Tenstorrent backend.
+# The `backend="tt"` argument explicitly selects TT-XLA.
+compiled_model = jax.jit(model, backend="tt")
```
[the remark this answers is not in the document]
Comment (einnesTT) [reply tt-forge/tt-xla/compile_models_single_chip.md:104]: Also, in your compile config file you sent me it showed something that appears to convert the options for use with JAX. Should I add that in? Does this make options available in JAX? 

```
def to_jax_compiler_options(self) -> Dict[str, str]:
        """
        Convert CompilerConfig to JAX compiler_options dictionary format.

        Returns:
            Dictionary of compiler options in the format expected by jax.jit()
        """
        options = {}

        if self.enable_optimizer:
            options["enable_optimizer"] = "true"

        if self.enable_memory_layout_analysis:
            options["enable_memory_layout_analysis"] = "true"

        if self.enable_l1_interleaved:
            options["enable_l1_interleaved"] = "true"

        if self.enable_bfp8_conversion:
            options["enable_bfp8_conversion"] = "true"

        if self.enable_fusing_conv2d_with_multiply_pattern:
            options["enable_fusing_conv2d_with_multiply_pattern"] = "true"

        return options
 ```

File: tt-forge/tt-xla/compile_models_single_chip.md (hunk 2 above)
Comment (einnesTT) [reply tt-forge/tt-xla/compile_models_single_chip.md:85]: Updated.

File: tt-forge/tt-xla/compile_models_single_chip.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+output = output.to("cpu")
+```
+
+The following chart lists the aspects of the template that are Tenstorrent specific: 
+
+| Code | Explanation |
+|------|-------------|
+| `xr.set_device_type("TT")` | Sets the target device to Tenstorrent hardware ("TT"). |
+| `options = {...}` | Defines a dictionary of TT-XLA compiler flags.|
+| `"enable_optimizer": "true"` | A TT-XLA compiler flag, it goes in `options = {...}`. Enables optimizer passes in MLIR. This includes various optimizations such as improving tensor memory layouts, operation configurations etc. |
```
[the remark this answers is not in the document]
Comment (nvukobratTT) [reply tt-forge/tt-xla/compile_models_single_chip.md:79]: Different shareholders are introducing different compile options, so it's not a single point of contact. For starters, it'll be best to ping on the tt-forge or tt-xla channel. If there is no response, feel free to reach me directly :))

File: tt-forge/tt-xla/compile_models_single_chip.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+# The `backend="tt"` argument explicitly selects TT-XLA.
+compiled_model = jax.jit(model, backend="tt")
+
+# Run the compiled model.
+output = compiled_model(input_data)
+
+print("Output:", output)
+
+# Optional: serialize the compiled artifacts for deployment or inspection.
+serialize_compiled_artifacts_to_disk(model, input_data, output_prefix="output/model")
```
[the remark this answers is not in the document]
Comment (nvukobratTT) [reply tt-forge/tt-xla/compile_models_single_chip.md:112]: Just shared similar answer above :))

File: tt-forge/tt-xla/compile_models_single_chip.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (einnesTT) [reply tt-forge/tt-xla/compile_models_single_chip.md]: updated

File: tt-forge/tt-xla/compile_models_single_chip.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (einnesTT) [reply tt-forge/tt-xla/compile_models_single_chip.md]: For now I'm going to leave this since it's useful in the tutorial. I have a ticket to get the compile information together as requested and I can update this with a link at that time.
